### PR TITLE
Fix intermittent issues on CI

### PR DESCRIPTION
**Pull request by route**

### CODE REVIEW

**Comment by route** [comment]
@iainbeeston Tests look much better now, the real cause was https://github.com/machinio/cuprite/pull/36/files#diff-9948fab217d02bb27cd531aa7af30938R40

@jcoglan I see that in https://github.com/faye/websocket-driver-ruby/issues/46 people had similar issues. It looks like when you send message way too faster than `on_open` is called websocket connection remains indefinitely in :connecting state. I didn't find any better solution than put sleep 0.01
Don't think it's the best. What can we do?

File: lib/capybara/cuprite/browser/web_socket.rb [hunk 1]
```diff
@@ -51,9 +46,24 @@ def on_message(event)
...
+        @driver.text(json)
+        @logger&.puts("\n\n>>> #{json}")
+      end
+
       def write(data)
         @sock.write(data)
       end
+
+      def close
+        @driver.close
```
Comment (iainbeeston) [inline lib/capybara/cuprite/browser/web_socket.rb]: Should Socket#close also call `@thread.join`, just like Client#close above? When I was reading the examples for websocket-driver I noticed that they do that in their socket class but curprite does not

**Comment by iainbeeston** [comment]
Wow travis looks far better on this branch! 😀

File: lib/capybara/cuprite/browser/web_socket.rb [hunk 2]
```diff
@@ -51,9 +46,24 @@ def on_message(event)
         @logger&.puts("    <<< #{event.data}\n")
       end
 
+      def on_close(_event)
+        @messages.close
+        @thread.kill
+      end
+
+      def send(data)
```
[the remark this answers is not in the document]
Comment (route) [reply lib/capybara/cuprite/browser/web_socket.rb]: right, I was just copy-pasting from websocket example, I'll put it back

File: lib/capybara/cuprite/browser/web_socket.rb (hunk 1 above)
Comment (route) [reply lib/capybara/cuprite/browser/web_socket.rb]: `websocket-driver` has two versions of `close`. One is event-driven https://github.com/faye/websocket-driver-ruby/blob/master/examples/tcp_client.rb#L25 and another one is plain close like the one we use https://github.com/faye/websocket-driver-ruby/blob/master/examples/tcp_client.rb#L42. I think it's pretty much close to what they have in example. Though `@thread.join` doesn't help as thread can be blocked on IO and they have to use `kill` as we do.

**Comment by route** [comment]
@iainbeeston oook looks good enough to me to merge it and then deal with refactoring and stuff

**Pull request merged**